apis/url.go

```diff
@@ -41,6 +41,30 @@ func ParseURL(u string) (*URL, error) {
 	return (*URL)(pu), nil
 }
 
+// HTTP creates an http:// URL pointing to a known domain.
+func HTTP(domain string) *URL {
+	return &URL{
+		Scheme: "http",
+		Host:   domain,
+	}
+}
+
+// HTTPS creates an https:// URL pointing to a known domain.
+func HTTPS(domain string) *URL {
+	return &URL{
+		Scheme: "https",
+		Host:   domain,
+	}
+}
+
+// IsEmpty returns true if the URL is `nil` or represents an empty URL.
+func (u *URL) IsEmpty() bool {
```

Comment:
how's is it possible for u to be nil below because you have to call through it? Am I confused again?

Reply:
You can call on a nil receiver (as long as you don't access any field members).

Reply:
Ack, thanks :)